source/features/show-associated-branch-prs-on-fork.tsx

```diff
@@ -0,0 +1,112 @@
+import React from 'dom-chef';
+import select from 'select-dom';
+import cache from 'webext-storage-cache';
+import mergeIcon from 'octicon/git-merge.svg';
+import pullRequestIcon from 'octicon/git-pull-request.svg';
+import * as api from '../libs/api';
+import features from '../libs/features';
+import * as pageDetect from '../libs/page-detect';
+import observeElement from '../libs/simplified-element-observer';
+import {getRepoURL, getRepoGQL, getOwnerAndRepo, getUsername} from '../libs/utils';
+
+interface PullRequest {
+	number: number;
+	state: string;
+	isDraft: boolean;
+	url: string;
+}
+
+const getOpenPullRequests = cache.function(async (): Promise<Record<string, PullRequest>> => {
+	const {repository} = await api.v4(`
+		repository(${getRepoGQL()}) {
+			refs(refPrefix: "refs/heads/", last: 100) {
+				nodes {
+					name
+					associatedPullRequests(last: 1, orderBy: {field: CREATED_AT, direction: DESC}) {
+						nodes {
+							number
+							state
+							isDraft
+							url
+						}
+					}
+				}
+			}
+		}
+	`);
+
+	const pullRequests = {};
+	for (const branches of repository.refs.nodes) {
+		if (branches.associatedPullRequests.nodes.length > 0) {
+			pullRequests[branches.name] = branches.associatedPullRequests.nodes[0];
```

Comment:
1 of 3: If you normalize the data here you won't have to normalize multiple times later.

```suggestion
	for (const {associatedPullRequests} of repository.refs.nodes) {
		if (associatedPullRequests.nodes.length > 0) {
			const [prInfo] = associatedPullRequests.nodes;
			prInfo.state = prInfo.isDraft ? 'Draft' : upperCaseFirst(prInfo.state);
			pullRequests[branches.name] = prInfo;
```